NixOS: Add Night Light Dependency and Enable Required Services

randibudi opened the pull request:
This PR adds night light functionality and fixes power management issues for NixOS users:

  ### Changes

  - Added wlsunset dependency for night light feature
  - Added NixOS module that auto-enables required services:
    - power-profiles-daemon (power profile switching)
    - upower (battery status monitoring)
  - Updated README with simplified installation

  ### Problems solved

  - Night light now works out of the box
  - Power profiles now function properly
  - Battery status now displays correctly
  - Installation simplified with single module import

Ly-sec commented:
In theory you can use matugen but it does involve some tinkering :p.
For Stylix, that could be a cool thing but since we can't test it, we have to rely on someone else to make it work; so if you want to do that, be my guest :heart: 

We got some merge conflicts that need to be resolved by the way.

ItsLemmy commented:
@idlip I have sent you an email. Let me know what you think. thanks

idlip commented:
> @idlip I have sent you an email. Let me know what you think. thanks

Sent me an email?
I don't see any. Can you please describe more on this regard?

Ly-sec commented:
> Hi @Ly-sec ,   
> 
> I noticed the NixOS installation instructions were removed from the README. Was this intentional?   
> 
> Just wanted to confirm since the Arch instructions are still there.   
> 
> Thanks!   

We moved all the in-depth instructions into the docs, perhaps that should be more clear though.

randibudi commented:
> > Hi @Ly-sec ,
> > I noticed the NixOS installation instructions were removed from the README. Was this intentional?
> > Just wanted to confirm since the Arch instructions are still there.
> > Thanks!
> 
> We moved all the in-depth instructions into the docs, perhaps that should be more clear though.

Okay, I understand. Actually, I updated the NixOS installation method to use the new NixOS module approach. I removed it because of the conflict, but now the conflict is resolved.   

Is my PR ready to be merged?

ItsLemmy commented:
If you need to patch the documentation, the repo is here :)
https://github.com/noctalia-dev/noctalia-docs

randibudi commented:
Done @ItsLemmy @Ly-sec
